File: backend/gateway/service/esign.go
package service

import (
	"context"
)

type ESignService interface {
	CreateEsignFlow(ctx context.Context, caseID int64, title string, documentIDs []int64, signerIDs []int64, creatorID int64) (map[string]interface{}, error)
	GetEsignList(ctx context.Context, caseID int64, page, pageSize int, userID int64, role int32) ([]map[string]interface{}, int64, error)
	GetEsignDetail(ctx context.Context, esignID int64, userID int64) (map[string]interface{}, error)
	SignDocument(ctx context.Context, esignID int64, userID int64, verifyCode string, signatureData string) error
	RevokeEsignFlow(ctx context.Context, esignID int64, userID int64, reason string) error
	SendEsignVerifyCode(ctx context.Context, esignID int64, mobile string) error
	GetEsignProgress(ctx context.Context, flowID string) (map[string]interface{}, error)
	StoreSignedDocToBlockchain(ctx context.Context, esignID int64) (map[string]interface{}, error)
	NotifySignerProgress(ctx context.Context, flowID string, signerID int64, notifyType string) error
}

type BlockchainService interface {
	StoreEvidence(ctx context.Context, caseID int64, evidenceID string, evidenceType string, evidenceName string, evidenceHash string, flowID string, metadata string, creatorID int64) (map[string]interface{}, error)
	GetCertList(ctx context.Context, caseID int64, evidenceType string, page, pageSize int, userID int64, role int32) ([]map[string]interface{}, int64, error)
	GetCertDetail(ctx context.Context, certNo string) (map[string]interface{}, error)
	VerifyEvidence(ctx context.Context, certNo string) (map[string]interface{}, error)
	DownloadCert(ctx context.Context, certNo string) (map[string]interface{}, error)
	PublicVerify(ctx context.Context, certNo string) (map[string]interface{}, error)
}

File: frontend/admin/src/services/esign.ts
import { request } from '../utils/request';
import axios from 'axios';

export interface EsignFlow {
  id: string;
  flowId: string;
  caseId: number;
  caseNo: string;
  docTitle: string;
  docUrl: string;
  signedDocumentUrl: string;
  status: number;
  statusName: string;
  signerCount: number;
  signedCount: number;
  expireTime: string;
  fadadaFlowId: string;
  crossPageSeal: number;
  bcCertNo: string;
  bcTxId: string;
  bcOnChainTime: string;
  bcStatus: number;
  createdAt: string;
  signers: EsignSigner[];
  blockchainCert?: BlockchainCertInfo;
}

export interface EsignSigner {
  id: string;
  userId: number;
  userName: string;
  userPhone: string;
  signOrder: number;
  signStatus: number;
  signStatusName: string;
  signedAt: string;
  signatureUrl: string;
  fadadaSignUrl: string;
  notifyStatus: number;
  notifyStatusName: string;
  notifySentAt: string;
}

export interface BlockchainCertInfo {
  certNo: string;
  txId: string;
  blockHeight: number;
  onChainTime: string;
  certUrl: string;
  qrcodeUrl: string;
  verifyUrl: string;
  status: number;
  evidenceHash: string;
}

export interface CreateEsignFlowParams {
  caseId: number;
  docType: number;
  docTitle: string;
  docContent?: string;
  docUrl?: string;
  templateId?: number;
  signerIds: number[];
  expireHours?: number;
  needNotify?: boolean;
  notifyType?: string;
  crossPageSeal?: boolean;
}

export interface BlockchainCertificate {
  id: string;
  certNo: string;
  evidenceId: string;
  evidenceType: string;
  evidenceTypeName: string;
  evidenceName: string;
  evidenceHash: string;
  caseId: number;
  caseNo: string;
  caseTitle: string;
  flowId: string;
  txId: string;
  blockHeight: number;
  onChainTime: string;
  certUrl: string;
  qrcodeUrl: string;
  verifyUrl: string;
  status: number;
  statusName: string;
  metadata: string;
  createdBy: number;
  createdAt: string;
}

export const esignService = {
  createFlow: (caseId: number | string, params: CreateEsignFlowParams) =>
    request.post<EsignFlow>(`/dispute/${caseId}/esign`, params),

  getList: (caseId: number | string, params?: { status?: number }) =>
    request.get<EsignFlow[]>(`/dispute/${caseId}/esign`, { params }),

  getDetail: (caseId: number | string, flowId: string) =>
    request.get<EsignFlow>(`/dispute/${caseId}/esign/${flowId}`),

  signDocument: (caseId: number | string, flowId: string, data: { recordId: number; verifyCode?: string }) =>
    request.post(`/dispute/${caseId}/esign/${flowId}/sign`, data),

  revokeFlow: (caseId: number | string, flowId: string, reason: string) =>
    request.post(`/dispute/${caseId}/esign/${flowId}/revoke`, { reason }),

  sendVerifyCode: (caseId: number | string, flowId: string) =>
    request.post(`/dispute/${caseId}/esign/${flowId}/send-code`),

  getProgress: (caseId: number | string, flowId: string) =>
    request.get(`/dispute/${caseId}/esign/${flowId}/progress`),
};

export const blockchainService = {
  storeEvidence: (caseId: number | string, data: {
    evidenceId: string;
    evidenceType: string;
    evidenceName: string;
    evidenceHash: string;
    flowId?: string;
    metadata?: string;
  }) => request.post(`/dispute/${caseId}/blockchain/store`, { case_id: caseId, ...data }),

  getCertList: (caseId: number | string, params?: { evidenceType?: string; page?: number; pageSize?: number }) =>
    request.get<BlockchainCertificate[]>(`/dispute/${caseId}/blockchain/certs`, { params }),

  getCertDetail: (caseId: number | string, certNo: string) =>
    request.get<BlockchainCertificate>(`/dispute/${caseId}/blockchain/cert/${certNo}`),

  verifyEvidence: (caseId: number | string, certNo: string) =>
    request.get(`/dispute/${caseId}/blockchain/verify`, { params: { certNo } }),

  downloadCert: (caseId: number | string, certNo: string) =>
    request.get(`/dispute/${caseId}/blockchain/cert/${certNo}/download`),

  publicVerify: (certNo: string) =>
    axios.get(`/api/v1/public/blockchain/verify/${certNo}`).then(res => res.data),
};
